drivers/system_pathways/app/controllers/system_pathways/warehouse_reports/reports_controller.rb

```diff
@@ -0,0 +1,204 @@
+###
+# Copyright 2016 - 2022 Green River Data Analysis, LLC
+#
+# License detail: https://github.com/greenriver/hmis-warehouse/blob/production/LICENSE.md
+###
+
+module SystemPathways::WarehouseReports
+  class ReportsController < ApplicationController
+    include WarehouseReportAuthorization
+    include AjaxModalRails::Controller
+    include ArelHelper
+    include BaseFilters
+    before_action :require_can_access_some_version_of_clients!, only: [:details, :items]
+    before_action :set_report, only: [:show, :destroy, :details, :chart_data]
+
+    def index
+      @pagy, @reports = pagy(report_scope.diet.ordered)
+      @report = report_class.new(user_id: current_user.id)
+      @filter.default_project_type_codes = @report.default_project_type_codes
+      previous_report = report_scope.where(user_id: current_user.id).last
+      @filter.update(previous_report.options) if previous_report
+
+      # Make sure the form will work
+      filters
+    end
+
+    def show
+      @filter = @report.filter
+      @filter.update(filter_params[:filters].merge(coc_codes: @filter.coc_codes))
+      @pathways_chart = SystemPathways::PathwaysChart.new(report: @report, filter: @filter)
+      respond_to do |format|
+        format.html {}
+        format.xlsx do
+          filename = "#{@report.title&.tr(' ', '-')}-#{Date.current.strftime('%Y-%m-%d')}.xlsx"
+          headers['Content-Disposition'] = "attachment; filename=#{filename}"
+        end
+      end
+    end
+
+    def create
+      @report = report_class.new(
+        user_id: current_user.id,
+      )
+      @report.filter = @filter
+
+      if @filter.valid?
+        @report.save
+        ::WarehouseReports::GenericReportJob.perform_later(
+          user_id: current_user.id,
+          report_class: @report.class.name,
+          report_id: @report.id,
+        )
+        # Make sure the form will work
+        filters
+        respond_with(@report, location: @report.index_path)
+      else
+        @pagy, @reports = pagy(report_scope.ordered)
+        filters
+        render :index
+      end
+    end
+
+    def destroy
+      @report.destroy
+      respond_with(@report, location: @report.index_path)
+    end
+
+    def chart_data
+      @chart_data = @report.allowed_section(params[:chart])
+      respond_to do |format|
+        format.json do
+          data = case @chart_data
+          when 'equity'
+            equity = SystemPathways::Equity.new(report: @report, filter: @filter)
+            equity.chart_data(params[:demographic_breakdown])
+          when 'time'
+            time = SystemPathways::TimeChart.new(report: @report, filter: @filter)
+            time.chart_data(params[:demographic_breakdown])
+          else
+            raise 'unknown chart type'
+          end
```

Comment:
Maybe
```suggestion
          clazz = case @chart_data
          when 'equity'
            SystemPathways::Equity
          when 'time'
            time = SystemPathways::TimeChart
          else
            raise 'unknown chart type'
          end
          data = clazz.new(report: @report, filter: @filter).chart_data(params[:demographic_breakdown])
```
I got confused what was being returned where...